desktop/main/NodeConfig.ts

```diff
@@ -21,19 +23,89 @@ export const loadNodeConfig = async (): Promise<NodeConfig> =>
         )
     : {};
 
+const getCustomNodeConfigName = (name: string) => `node-config.${name}.json`;
+
+export const writeCustomNodeConfig = async (
+  netName: string,
+  config: string
+): Promise<NodeConfig> => {
+  const netConfigPath = path.join(
+    USERDATA_DIR,
+    getCustomNodeConfigName(netName)
+  );
+
+  await fs.writeFile(netConfigPath, config, {
+    encoding: 'utf8',
+  });
+
+  return JSON.parse(config);
+};
+
+export const loadOrCreateCustomConfig = async (
+  netName: string
+): Promise<NodeConfig> =>
+  existsSync(path.join(USERDATA_DIR, getCustomNodeConfigName(netName)))
+    ? fs
+        .readFile(path.join(USERDATA_DIR, getCustomNodeConfigName(netName)), {
+          encoding: 'utf8',
+        })
+        .then((res) =>
+          res.startsWith('{') ? JSON.parse(res) : TOML.parse(res)
+        )
+    : writeCustomNodeConfig(netName, '{}');
+
 const loadSmeshingOpts = async (nodeConfig) => {
   const id = generateGenesisIDFromConfig(nodeConfig);
   const opts = StoreService.get(`smeshing.${id}`);
   return safeSmeshingOpts(opts, id);
 };
 
-export const downloadNodeConfig = async (networkConfigUrl: string) => {
+export const isNodeConfigAndDiscoveryConfigEqual = (
+  nodeConfig,
+  discoveryConfig
+) => {
+  let equal = true;
+  R.mergeDeepWithKey(
+    (_, clientConfig, config) => {
+      if (R.equals(clientConfig, config)) {
+        return clientConfig;
+      }
+
+      equal = false;
+
+      return config;
+    },
+    nodeConfig,
+    discoveryConfig
+  );
+
+  return equal;
+};
+
+export const downloadNodeConfig = async (
+  netName: string,
+  networkConfigUrl: string
+) => {
   const nodeConfig = await fetchNodeConfig(networkConfigUrl);
-  // Copy smeshing opts from previous node config or replace it with empty one
-  nodeConfig.smeshing = await loadSmeshingOpts(nodeConfig);
+  const clientConfig = await loadNodeConfig();
+
+  // download and write config only if not exists
+  if (!R.isEmpty(clientConfig)) {
+    // keep initial smeshing options
+    nodeConfig.smeshing = await loadSmeshingOpts(nodeConfig);
+    // forward config to $nodeConfig.next
+    return nodeConfig;
+  }
+
+  const customNodeConfig = await loadOrCreateCustomConfig(netName);
```

Comment:
The logic is wrong here. Isn't it?

It will create `node-config.testnet-05.json` and use it only if there is no `node-config.json` yet. And, moreover, we don't need to read the current merged node config at all — we will have a custom one (with only fields that are changing) per network and the actual config coming from the discovery.

```suggestion
  const customNodeConfig = await loadOrCreateCustomConfig(netName);
```